src/main/java/com/okta/tools/helpers/ProfileHelper.java

```diff
@@ -31,17 +35,17 @@ public void createAwsProfile(AssumeRoleWithSAMLResult assumeResult, String crede
         credentialsHelper.updateCredentialsFile(credentialsProfileName, awsAccessKey, awsSecretKey, awsSessionToken);
     }
 
-    public String getProfileName(AssumeRoleWithSAMLResult assumeResult, String oktaProfile) {
+    public String getProfileName(AssumeRoleWithSAMLResult assumeResult) {
         String credentialsProfileName;
-        if (StringUtils.isNotBlank(oktaProfile)) {
-            credentialsProfileName = oktaProfile;
+        if (StringUtils.isNotBlank(environment.oktaProfile)) {
+            credentialsProfileName = environment.oktaProfile;
         } else {
             credentialsProfileName = assumeResult.getAssumedRoleUser().getArn();
-            if (credentialsProfileName.startsWith("arn:aws:sts::")) {
-                credentialsProfileName = credentialsProfileName.substring(13);
-            }
-            if (credentialsProfileName.contains(":assumed-role")) {
-                credentialsProfileName = credentialsProfileName.replaceAll(":assumed-role", "");
+            Matcher matcher = assumedRoleUserPattern.matcher(credentialsProfileName);
+            if (matcher.matches()) {
+                credentialsProfileName = matcher.group("roleName") + "_" + matcher.group("account");
+            } else {
+                credentialsProfileName = "temp";
```

Comment:
:seedling: use the "parser" to pull our the role name and account number and use them to build the role name.

Having the account alias would be nice, but it's a bit clumsy to thread that from the role selection menu to here. Baby steps to fix the bug then polish later.